qualtran/surface_code/algorithm_specs.py

```diff
@@ -0,0 +1,40 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import attr
+from attrs import frozen
+
+_PRETTY_FLOAT = attr.ib(type=float, default=0.0, repr=lambda x: f'{x:g}')
+
+
+@frozen
+class AlgorithmSpecs:
```

Comment:
spec? that's short for specification. This doesn't really specify an algorithm. These are properties of an algorithm

Reply:
is this class redundant with `magic_state_factory.MagicStateCount`? Can you refactor to remove code duplication